kolena/_utils/krequests.py

```diff
@@ -53,16 +54,27 @@
 MAX_RETRIES = Retry(total=3, connect=3, read=0, redirect=0, status=0, backoff_factor=2)
 
 
+class JWTAuth(requests.auth.AuthBase):
+    """Attaches JWT Authorization to the given Request object"""
+
+    def __init__(self, jwt: Optional[str]):
+        self.jwt = jwt
+
+    def __call__(self, r):
+        r.headers["Authorization"] = f"Bearer {self.jwt}"
```

Comment:
Should this still set the auth header when `self.jwt` is absent? Should `JWTAuth` even be instantiated when the `jwt` is `None`?

Reply:
Since `_with_default_kwargs` is decorated with `@kolena_initialized`, it looks like `jwt` will always be present (`str` instead of `Optional[str]`). If that's the case, let's update the types on `JWTAuth.__init__` to reflect this non-nullability

Reply:
Yup, nice catch. Updated `JWTAuth.__init__` type hints.
Unfortunately, Pyright isn't as smart and flags a false negative due to https://github.com/kolenaIO/kolena/blob/trunk/kolena/_utils/state.py#L56